index2.js

```diff
@@ -65,49 +65,54 @@ function guess(min, max) {
 
 
 async function playComp() {  //computer guesses human number
-    console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
-
+    console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.");
+    max = await ask("First off, lets pick a range to guess from.  We will start with a minimum of 1.\nWhat should our maximum be?\n");
+    max = await isInteger(max);  //ensures max is an integer
     let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
     myNum = await isInteger(secretNumber);
+    while (myNum > max) {  //ensures player picks a number within range
+        myNum = await ask(`Please enter a number less than ${max}...\n`);
+        myNum = await isInteger(myNum);
+    }
     let numGuess = min + Math.floor(Math.random() * (max - min + 1)); //First guess should be random to not give away our secret
```

Comment:
Should the first guess be random? Wouldn't it be smarter for the computer to pick the middle number of the range for its first guess e.g. for 1-100, the first guess should be 50.

Reply:
Technically yes it totally should just be the middle of the range.... I just figured for the sake of gameplay, having the first guess always be dead center of the middle of the range gives away our guessing algorithm quicker than the game already does.  That's really the only reason I set the first guess at random.